executor/aggfuncs/window_func_test.go

```diff
@@ -69,6 +81,37 @@ func (s *testSuite) testWindowFunc(c *C, p windowTest) {
 	finalFunc.ResetPartialResult(finalPr)
 }
 
+func buildWindowMemTester(funcName string, tp byte, constantArg uint64, orderByCols int, numRows int, allocMemDelta int64, updateMemDeltaGens updateMemDeltaGens, isDistinct bool) windowMemTest {
+	windowTest := buildWindowTester(funcName, tp, constantArg, orderByCols, numRows)
+	pt := windowMemTest{
+		windowTest:         windowTest,
+		allocMemDelta:      allocMemDelta,
+		updateMemDeltaGens: updateMemDeltaGens,
+		isDistinct:         isDistinct,
+	}
+	return pt
+}
+
+func (s *testSuite) testWindowMemFunc(c *C, p windowMemTest) {
+	srcChk := p.windowTest.genSrcChk()
+
+	desc, err := aggregation.NewAggFuncDesc(s.ctx, p.windowTest.funcName, p.windowTest.args, false)
+	c.Assert(err, IsNil)
+	finalFunc := aggfuncs.BuildWindowFunctions(s.ctx, desc, 0, p.windowTest.orderByCols)
+	finalPr, memDelta := finalFunc.AllocPartialResult()
+	c.Assert(memDelta, Equals, p.allocMemDelta)
+
+	updateMemDeltas, err := p.updateMemDeltaGens(srcChk, p.windowTest.dataType)
+	c.Assert(err, IsNil)
+	iter := chunk.NewIterator4Chunk(srcChk)
+	i := 0
+	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
+		memDelta, _ = finalFunc.UpdatePartialResult(s.ctx, []chunk.Row{row}, finalPr)
```

Comment:
I think we should check for errors here.

Reply:
Sure. I'll take a look. Thanks!